src/translate.py

```diff
@@ -1062,6 +1067,9 @@ def late_field_path(self) -> Optional[AccessPath]:
                 self.field_path = field_path
                 self.type.unify(field_type)
 
+                # Check invariant on self.field_path
+                self.__post_init__()
```

Comment:
Is this weird? (I could make the invariant check a separate fn?)

Reply:
Looks rather goofy. I don't mind terribly but probably best to put it in a separate function yeah